tests/std/tests/P0355R7_calendars_and_time_zones_time_zones/test.cpp

```diff
@@ -91,9 +117,268 @@ void timezone_names_test() {
+        assert(tz->to_local(to_standard + minutes{30}) == Day_to_Std.second.local_begin() + minutes{30});
+        assert(tz->to_local(to_standard - minutes{30}) == Day_to_Std.first.local_end() - minutes{30});
+
+        const auto& to_daylight = Std_to_Day.first.end();
+        assert(tz->to_local(to_daylight) == Std_to_Day.second.local_begin());
+        assert(tz->to_local(to_daylight + minutes{30}) == Std_to_Day.second.local_begin() + minutes{30});
+        assert(tz->to_local(to_daylight - minutes{30}) == Std_to_Day.first.local_end() - minutes{30});
+    }
+    {
+        using namespace LA;
+        auto tz = tzdb.locate_zone(Tz_name);
+        assert(tz != nullptr);
+
+        const auto& to_standard = Day_to_Std.first.end();
+        assert(tz->to_local(to_standard) == Day_to_Std.second.local_begin());
+        assert(tz->to_local(to_standard + minutes{30}) == Day_to_Std.second.local_begin() + minutes{30});
+        assert(tz->to_local(to_standard - minutes{30}) == Day_to_Std.first.local_end() - minutes{30});
+
+        const auto& to_daylight = Std_to_Day.first.end();
+        assert(tz->to_local(to_daylight) == Std_to_Day.second.local_begin());
+        assert(tz->to_local(to_daylight + minutes{30}) == Std_to_Day.second.local_begin() + minutes{30});
+        assert(tz->to_local(to_daylight - minutes{30}) == Std_to_Day.first.local_end() - minutes{30});
+    }
+}
+
+void assert_local(const time_zone* tz, local_seconds time, int result, const sys_info& first, const sys_info& second) {
+    const auto info = tz->get_info(time);
+    assert(info.result == result);
+    assert(info.first == first);
+    assert(info.second == second);
+
+    // time_zone::to_sys depends heavily on local_info so just test it here
+    // to exhaust all corner cases.
+    sys_seconds sys_earliest{time.time_since_epoch() - info.first.offset};
+    sys_seconds sys_latest{time.time_since_epoch() - info.second.offset};
+    try {
+        assert(tz->to_sys(time) == sys_earliest);
+        assert(result == local_info::unique);
+    } catch (nonexistent_local_time ex) {
+        assert(result == local_info::nonexistent);
+    } catch (ambiguous_local_time ex) {
+        assert(result == local_info::ambiguous);
+    }
+
+    if (result == local_info::unique) {
+        assert(tz->to_sys(time, choose::earliest) == sys_earliest);
+        assert(tz->to_sys(time, choose::latest) == sys_earliest);
+    } else if (result == local_info::nonexistent) {
+        assert(tz->to_sys(time, choose::earliest) == info.first.end);
+        assert(tz->to_sys(time, choose::latest) == info.first.end);
+    } else if (result == local_info::ambiguous) {
+        assert(tz->to_sys(time, choose::earliest) == sys_earliest);
+        assert(tz->to_sys(time, choose::latest) == sys_latest);
+    }
+}
+
+void validate_get_local_info(const time_zone* tz, const std::pair<Transition, Transition>& transition, int result) {

```

Comment:
Should include `<utility>` for `pair`, and should be unqualified (this test is already `using namespace std;`).